### PR TITLE
[iOS] Automatically focus search bar when navigating to search screen

### LINKED ISSUE #525: Search Feature: focus on search bar when transitioning search screen
## Overview
Currently, when user transition to search screen, search bar is not focused on.
But users transition because of searching sessions, so it is better for the UX to be focused on the search bar from the beginning.

## Image
When transitioning to search screen...
| As Is | To Be |
| ---- | ---- |
|  |  |

### PR DESCRIPTION
**Pull request by nyagasan**
## Issue
close #525 

## Overview (Required)
- Use `.searchable`'s `isPresented` parameter to automatically focus the search bar.

## Links
- 

## Screenshot (Optional if screenshot test is present or unrelated to UI)
When transitioning to search screen
As Is | After
:--: | :--:
 | 

## Movie (Optional)
Before | After
:--: | :--:
<video src="" width="300" > | <video src="" width="300" >

### CODE REVIEW

File: app-ios/Sources/SearchFeature/SearchView.swift [hunk 1]
```diff
@@ -50,11 +51,15 @@ public struct SearchView: View {
...
             ),
+            isPresented: $isfocused,
             placement: .navigationBarDrawer(displayMode: .always)
         )
         .foregroundStyle(AssetColors.Surface.onSurface.swiftUIColor)
         .onAppear {
             store.send(.view(.onAppear))
+            DispatchQueue.main.async {
+                isfocused = true
+            }
```
Comment (ry-itto) [inline app-ios/Sources/SearchFeature/SearchView.swift:62]: SwiftUI.View is isolated on MainActor, so I think run this code on main thread isn't needed 👀 
How about following?

```suggestion
            isfocused = true
```

**Review by ry-itto — COMMENTED** [review]
@nyagasan 
Thank you for contributing!
I left a review comment, please check it!

File: app-ios/Sources/SearchFeature/SearchView.swift (hunk 1 above)
Comment (nyagasan) [reply app-ios/Sources/SearchFeature/SearchView.swift:62]: @ry-itto 
Thank you for comment.
I have added a delay using `DispatchQueue` after reading this article:

https://www.rk-k.com/archives/7925

If the issue is resolved with the latest SwiftUI.View as mentioned in the comments, I am considering changing to the suggested approach.

Comment (shin-usu) [reply app-ios/Sources/SearchFeature/SearchView.swift:62]: Doesn't it actually work without Dispatch?👀
If so, I think there should be comment for this issue.

File: app-ios/Sources/SearchFeature/SearchView.swift [hunk 2]
```diff
@@ -8,6 +8,7 @@ import Theme
 
 public struct SearchView: View {
     @Bindable private var store: StoreOf<SearchReducer>
+    @State private var isfocused: Bool = false
```
[the remark this answers is not in the document]
Comment (shin-usu) [reply app-ios/Sources/SearchFeature/SearchView.swift:11]: Oh, I'm sorry. Probably, FocusState isn't able to be used for searchable modifier.
Don't mind this comment.

[the remark this answers is not in the document]
Comment (shin-usu) [reply app-ios/Sources/SearchFeature/SearchView.swift:11]: But I think following is correct case 👀 
```suggestion
    @State private var isFocused: Bool = false
```

File: app-ios/Sources/SearchFeature/SearchView.swift (hunk 1 above)
Comment (nyagasan) [reply app-ios/Sources/SearchFeature/SearchView.swift]: Thank you for your feedback. After testing on the simulator, everything worked without any issues. Therefore, I have removed the Dispatch.

**Review by shin-usu — APPROVED** [review]
LGTM🎉

**Pull request merged**